### PR TITLE
EngiMarket Part 5 - Prototype Canister

### PR DESCRIPTION
**Pull request by Robustin**
Relevant video, indoor fulton sold seperately: https://youtu.be/9MS1TWSqM3E

P.S. Fuck TGUI tho

:cl: Robustin
add: Added the prototype canister with expanded volume, valve pressure, and access/timer features
/:cl: #

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Telescience, where u @

**Comment by FantasticFwoosh** [comment]
Infinite flame throwing with dark purple slimes.

**Comment by kevinz000** [comment]
ehh good idea standalone but still don't like your points.

**Comment by jughu** [comment]
Buffing canisterbombs

**Comment by Grazyn** [comment]
Add a maint panel so you can hack it to increase max pressure, attach signallers to "open valve" wire etc.

**Comment by Robustin** [comment]
@Grazyn If you get around to playing with this you'll find that expanded volume makes the valve setting even more powerful than what youre used to, 

As for signallers and such, I honestly tried, it was my first idea instead of having to fuck with TGUI, unfortunately I don't have much experience with wires so I just felt an innate timer would be better anyway.

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 1]
```diff
@@ -344,6 +421,26 @@
...
+		if("timer")
+			var/change = params["change"]
+			if(change == "reset")
+				timer_set = default_timer_set
+			else if(change == "decrease")
+				timer_set = max(minimum_timer_set, timer_set - 10)
+			else if(change == "increase")
+				timer_set = min(maximum_timer_set, timer_set + 10)
+			else if(change == "input")
+				var/user_input = input(usr, "Set time to valvle toggle.", name) as null|num
```
Comment (Militaires) [inline code/modules/atmospherics/machinery/portable/canister.dm]: valve

File: code/modules/atmospherics/machinery/portable/canister.dm [hunk 2]
```diff
@@ -39,6 +49,13 @@
 		"caution" = /obj/machinery/portable_atmospherics/canister,
 	)
 
+/obj/machinery/portable_atmospherics/canister/interact(mob/user)
+	if(!allowed(user))
+		user << "<span class='warning'>Error - Unauthorized User</span>"
```
Comment (AnturK) [inline code/modules/atmospherics/machinery/portable/canister.dm]: to_chat

**Review by AnturK — COMMENTED** [review]
Just some minor issue

**Comment by lzimann** [comment]
You still need to commit both compiled tgui files.

**Pull request merged**